### PR TITLE
HIPP-1591: Add new OAS validation endpoint

### PR DESCRIPTION
**Pull request by victorarbuesmallada**

### CODE REVIEW

File: test/uk/gov/hmrc/apihubapplications/controllers/OASControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    "must return BadRequest when the request is invalid" in {
+      val fixture = buildFixture()
+
+      running(fixture.application) {
+        val request = FakeRequest(routes.OASController.validateOAS())
+          .withBody(Json.parse("""["badrequest"]"""))
+
+        val result = route(fixture.application, request).value
+
+        status(result) mustBe BAD_REQUEST
```
Comment (llmikeyj) [inline test/uk/gov/hmrc/apihubapplications/controllers/OASControllerSpec.scala:114]: I think there could be a response body with the bad request, do we need to check that the controller will pass this on?

Comment (victorarbuesmallada) [reply test/uk/gov/hmrc/apihubapplications/controllers/OASControllerSpec.scala:114]: in this case, we don't return the payload request parsing error but just log it when it happens, as in this other [example](https://github.com/hmrc/api-hub-applications/blob/main/test/uk/gov/hmrc/apihubapplications/controllers/AccessRequestsControllerSpec.scala#L79).
Happy to have a discussion if that needs to change.

Comment (llmikeyj) [reply test/uk/gov/hmrc/apihubapplications/controllers/OASControllerSpec.scala:114]: I think we also handle bad requests with bodies though,

https://github.com/hmrc/api-hub-applications/blob/main/app/uk/gov/hmrc/apihubapplications/connectors/APIMConnectorImpl.scala#L159

Do we need to pass this body to the front end?

Comment (llmikeyj) [reply test/uk/gov/hmrc/apihubapplications/controllers/OASControllerSpec.scala:114]: I agree, but are we interested in showing this kind of error to the user? One for Olesegun maybe?

**Review by llmikeyj — APPROVED** [review]
:LGTM.

**Pull request merged**